ui.apps/src/main/content/jcr_root/apps/asset-share-commons/clientlibs/clientlib-site/js/storage.js

```diff
@@ -0,0 +1,262 @@
+        var storage = _getLocalStorage(),
+            assets;
+
+        if(storage["users"] && storage["users"][currentUserId]) {
+
+            storage["users"][currentUserId]["cart"] = storage["users"][currentUserId]["cart"] || {};
+            assets = storage["users"][currentUserId]["cart"]["assets"] || [];
+
+            if(assets.indexOf(assetPath) < 0) {
+                assets.push(assetPath);
+                storage["users"][currentUserId]["cart"]["assets"] = assets;
+                _updateLocalStorage(storage);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /* removes an asset path from the cart and persists to local storage */
+    function removeCartAsset(assetPath) {
+        var storage = _getLocalStorage(),
+            assets,
+            index;
+
+        if(storage["users"] && storage["users"][currentUserId] && storage["users"][currentUserId].cart) {
+            assets = storage["users"][currentUserId]["cart"]["assets"] || [];
+
+            index = assets.indexOf(assetPath);
+            if( index >= 0) {
+                assets.splice(index,1);
+                storage["users"][currentUserId]["cart"]["assets"] = assets;
+                _updateLocalStorage(storage);
+                return true;
+            }
+        }
+        return false;
+    }
+
+     /* clears the cart and sets it to an empty array */
+     function clearCartAssets() {
+        var storage = _getLocalStorage();
+
+        if(storage["users"] && storage["users"][currentUserId]) {
+            storage["users"][currentUserId]["cart"] = storage["users"][currentUserId]["cart"] || {};
+            storage["users"][currentUserId]["cart"]["assets"] = [];
+
+            _updateLocalStorage(storage);
+            return true;
+        }
+
+        return false;
+    }
+
+    /* Gets the contents of the cart Array, if not set an empty array is returned */
+    function getCartAssets() {
+        var storage,
+            paths = [];
+
+        storage = _getLocalStorage();
+        if(storage["users"] && storage["users"][currentUserId] && storage["users"][currentUserId].cart) {
+            paths = storage["users"][currentUserId].cart.assets;
+        }
+
+        return paths;
+    }
+
+    /* Get status if profile ready and storage can read from local storage */
+    function isReady() {
+        return localStorageEnabled && profileLoaded;
+    }
+
+    /* Check if browser supports local storage capabilities */
+    function _localStorageCheck() {
+        var storage;
+
+        if(localStorageEnabled) {
+            return true;
+        }
+
+        try {
+            storage = window.localStorage;
+            var x = '__storage_test__';
+            storage.setItem(x, x);
+            storage.removeItem(x);
+            localStorageEnabled = true;
+            return true;
+        }
+        catch(e) {
+            localStorageEnabled = false;
+            return e instanceof DOMException && (
```

Comment:
If this errors, what happens? Does the user get any indication of something being wrong?

Reply:
no. Will look at the other refactors and then try and figure something out. For ContextHub I believe it had some built-in fallbacks to support session storage if Local storage is not available. Not sure if we need to go that far or if a simple warning/error is displayed...